framework/src/problems/FEProblemBase.C

```diff
@@ -205,8 +205,18 @@ FEProblemBase::validParams()
                                        rz_coord_axis,
                                        "The rotation axis (X | Y) for axisymetric coordinates",
                                        "Please use 'Mesh/rz_coord_axis' instead");
-  params.addParam<bool>(
-      "kernel_coverage_check", true, "Set to false to disable kernel->subdomain coverage check");
+  MooseEnum kernel_coverage_check_modes("TRUE ON FALSE OFF SKIP_LIST ONLY_LIST", "TRUE");
+  params.addParam<MooseEnum>(
+      "kernel_coverage_check",
+      kernel_coverage_check_modes,
+      "Controls, if and how a kernel subdomain coverage check is performed. With 'TRUE' or 'ON' "
+      "all subdomains are checked (the default). Setting 'FALSE' or 'OFF' will disable the check "
+      "for all subdomains. To exclude a predefined set of subdomains 'SKIP_LIST' is to be used, "
+      "while the subdomains to skip are to be defined in the parameter "
+      "'kernel_coverage_block_list'. To limit the check to a list of subdomains, 'ONLY_LIST' is to "
+      "be used (again, using the parameter 'kernel_coverage_block_list').");
+  params.addParam<std::vector<SubdomainName>>(
+      "kernel_coverage_block_list", {}, "List of subdomains for kernel coverage check");
```

Comment:
Mention in the doc string that this can be a list of blocks to check or blocks to skip the check